### DISCUSSION TITLE
[W3HW3] 判斷質數

### DISCUSSION THREAD

**Discussion by mhsiungw** [Lidemy/forum#81] [程式導師實驗計畫]
```
var readline = require('readline');
var rl = readline.createInterface({
  input: process.stdin
});

var lines = []

// 讀取到一行，先把這一行加進去 lines 陣列，最後再一起處理
rl.on('line', function (line) {
  lines.push(line)
});

// 輸入結束，開始針對 lines 做處理
rl.on('close', function() {
  solve(lines)
})


function solve (lines){
	for (let j = 1 ; j <= lines ; j ++){
		isPrimeNumber(j)
	}
}


function isPrimeNumber (n){
	let sum = ''
	if (n === 1) {
		console.log('Composite')
	} else {
		for (let i = 1 ; i <= n ; i ++ ) {
			if (n % i === 0) {
				sum += i
			}	
		} if (sum.length <= 2) {
			console.log('Prime')
		} else {
			console.log('Composite')
		}
	}
}

```

大家好，我用這段程式碼在LIOJ提交第三週的**判斷質數**作業，但一直得到 WA，是我的輸入有問題嗎？
因為我的理解是輸入一個數字Ｎ，判斷所有小於Ｎ的數字是否為質數，
麻煩大家解惑了！ 
多謝

**Comment by allenliaooooooo** [comment]
輸入的數不一定是從 1 到 N 之間的數
N 只是會有幾個需要判斷的數而已
然後資料型態要注意哦

**Reply by mhsiungw** [reply]
謝謝你的回覆！所以這題是因為我的輸入方式錯誤才無法過 LIOJ 嗎？
我的 ```solve(lines)``` 的 ```lines``` 必須放有幾個要判的字嗎？

另外，資料型態的部分我有點困惑，是指 ```sum = ''``` 那一部分嗎？

再次謝謝你的幫忙ＱＡＱ

**Reply by allenliaooooooo** [reply]
你有在自己 local 端先測試過嗎？
可以試著把 lines 印出來看看長什麼樣子
再思考要怎麼處理 lines ～

**Reply by mhsiungw** [reply]
有，程式碼是正常的，確實可以判斷是不是質數。
只是就卡在輸入方式不知道該怎麼設定，我再研究研究，謝謝你！

**Reply by nuoxoxo** [reply]
是輸入的資料型態的問題。你可以將 lines 這個陣列 log 出來：
`[ '5', '1', '2', '3', '4', '5' ]`
是一個字串陣列。你要把傳進 isPrimeNumber 的 n 轉化成`Number`

**Reply by mhsiungw** [reply]
@nuoxoxo 好，謝謝你！！
